docs/contributing/999-rfcs/2024-06-02-ts-client.md

```diff
@@ -0,0 +1,121 @@
+---
+title: "TypeScript Clients Across Wing Applications"
+description: TypeScript Clients Across Wing Applications
+---
+
+# TypeScript Clients Across Wing Applications
+
+- **Author(s)**: @MarkMcCulloh
+- **Submission Date**: 2024-06-02
+
+## Summary
+
+Allow for the generation of TypeScript clients that can interact with wing-defined infrastructure.
+Unlike `extern` or `@inflight`, the client is meant to be consumed by TypeScript applications rather than directly executed by Wing itself.
+
+The goal of this document is to walk through the DX of this experience.
+
+An important consideration for this discussion is allowing the generated client to reference resources defined from a separate wing application.
+To facilitate this separation, it seems necessary to introduce a mechanism to export/import deployed data between Wing apps.
+The rest of this proposal does it via `cloud.Outputs`, but that likely deserves its own RFC.
+
+## Example Situation
+
+Acme Co. has a platform team and several application teams. Their platform team manages Kubernetes clusters in AWS with additional AWS resources as needed.
+Application teams write applications in TypeScript and deploy them to these shared Kubernetes clusters.
+The centralized infrastructure definitions are written in Wing in their own repository.
+The applications are written in TypeScript but also have Wing files for infrastructure/deployment.
```

Comment:
> but also have Wing files for infrastructure/deployment.

What's that? Can you elaborate?

Reply:
I was expecting wing to be used to deploy the application, because otherwise it's not clear to me how (or why) wing fits into the workflow.

Reply:
In this workflow wing will only be used to define the preflight (infra) while the app will be deployed by other tools like ArgoCD for example.